docsrc/source/api/api_docs/index.rst

```diff
@@ -17,30 +15,91 @@ API Docs
+
+core
+=====
+
+- :ref:`CoreConfig<ug-CoreConfig>`: Module to contain configurations of the optimization process.
 - :ref:`quantization_config<ug-quantization_config>`: Module to configure the quantization process.
 - :ref:`mixed_precision_quantization_config<ug-mixed_precision_quantization_config>`: Module to configure the quantization process when using mixed-precision PTQ.
-- :ref:`network_editor<ug-network_editor>`: Module to edit your model during the quantization process.
-- :ref:`target_platform<ug-target_platform>`: Module to create and model hardware-related settings to optimize the model according to, by the hardware the optimized model will use during inference.
+- :ref:`network_editor<ug-network_editor>`: Module to modify the optimization process for troubleshooting.
+- :ref:`FolderImageLoader<ug-FolderImageLoader>`: Class to use an images directory as a representative dataset.
+- :ref:`pytorch_kpi_data<ug-pytorch_kpi_data>`: A function to compute KPI data that can be used to calculate the desired target KPI for PyTorch models.
+- :ref:`keras_kpi_data<ug-keras_kpi_data>`: A function to compute KPI data that can be used to calculate the desired target KPI for Keras models.
+
+
+data_generation
+=================
+
+- :ref:`pytorch_data_generation_experimental<ug-pytorch_data_generation_experimental>`: A function to generate data for a Pytorch model (experimental).
+- :ref:`get_pytorch_data_generation_config<ug-get_pytorch_data_generation_config>`: A function to load a DataGenerationConfig for Pytorch data generation (experimental).
+- :ref:`keras_data_generation_experimental<ug-keras_data_generation_experimental>`: A function to generate data for a Keras model (experimental).
+- :ref:`get_keras_data_generation_config<ug-get_keras_data_generation_config>`: A function to generate a DataGenerationConfig for Tensorflow data generation (experimental).
+
+pruning
+===========
 
-Classes
+- :ref:`pytorch_pruning_experimental<ug-pytorch_pruning_experimental>`: A function to apply structured pruning for Pytorch models (experimental).
+- :ref:`keras_pruning_experimental<ug-keras_pruning_experimental>`: A function to apply structured pruning for Keras models (experimental).
+
+- :ref:`PruningConfig<ug-PruningConfig>`: Configuration for the pruning process.
+- :ref:`PruningInfo<ug-PruningInfo>`: Information about the pruned model such as pruned channel indices, etc.
+
+
+exporter
 =========
-- :ref:`GradientPTQConfig<ug-GradientPTQConfig>`: Class to configure GradientPTQC options for gradient based post training quantization.
-- :ref:`FolderImageLoader<ug-FolderImageLoader>`: Class to use an images directory as a representative dataset.
-- :ref:`FrameworkInfo<ug-FrameworkInfo>`: Class to wrap framework information to be used by MCT when optimizing models.
+
+- :ref:`exporter<ug-exporter>`: Module that enables to export a quantized model in different serialization formats.
```

Comment:
They are detailed in this link (which I think is correct since there are examples that need this context), so I do not think it should be duplicated. Maybe I can move them here instead, but I think we shouldn't.